app/controllers/spree/api_helpers_decorator.rb

```diff
@@ -0,0 +1,4 @@
+Spree::Api::ApiHelpers.class_eval do
+  #TIHI
+  class_variable_set('@@payment_attributes', class_variable_get('@@payment_attributes') + [:braintree_device_data])
```

Comment:
Don't really like this, but all my other solutions involved really convoluted decorators for the solidus checkout controllers, this seems to be an all right solution while being one of the things at the top of of my "DON'T EVER DO THIS" list :\

Reply:
I think as long as class vars are set at boot time and touched never again, they are reasonably safe.